libs/compiler/mipsgen.c

```diff
@@ -273,6 +273,18 @@ typedef struct lvalue
 	const item_t type;						/**< Value type */
 } lvalue;
 
+typedef struct mutable_lvalue
+{
+	lvalue_kind_t kind;						/**< Value kind */
+	mips_register_t base_reg;				/**< Base register */
+	union									/**< Value location */
+	{
+		item_t reg_num;						/**< Register where the value is stored */
+		item_t displ;						/**< Stack displacement where the value is stored */
+	} loc;
+	item_t type;							/**< Value type */
+} mutable_lvalue;
```

Comment:
The `lvalue` struct had all its field marked as const just to make sure they are not modified by accident. If you really need them to be mutable (seems like it), just remove modifiers